src/envoy/http/service_control/client_cache.cc

```diff
@@ -442,18 +452,20 @@ void ClientCache::callQuota(const AllocateQuotaRequest& request,
 void ClientCache::handleQuotaOnDone(const Status& http_status,
                                     AllocateQuotaResponse* response,
                                     QuotaDoneFunc on_done) {
+  QuotaResponseInfo response_info;
   if (http_status.ok()) {
-    QuotaResponseInfo response_info;
-    Status quota_status = ::espv2::api_proxy::service_control::RequestBuilder::
-        ConvertAllocateQuotaResponse(*response, config_.service_name(),
-                                     &response_info);
+    Status quota_status = ::espv2::api_proxy::service_control::
+        CheckResponseConverter::ConvertAllocateQuotaResponse(
+            *response, config_.service_name(), &response_info);
 
     collectScResponseErrorStats(response_info.error_type);
-    on_done(quota_status);
+    on_done(quota_status, response_info);
   } else {
     // Most likely an auth error in ESPv2 or API producer deployment.
     filter_stats_.filter_.denied_producer_error_.inc();
-    on_done(http_status);
+    response_info.error_name = kScCallFail5XX;
```

Comment:
This could either be a 4xx or 5xx, you need some logic to differentiate between it. Or we could just put the full status code, or we could just generalize to `http_call_fail`.